tests/Message/ApplePayAuthorizeRequestTest.php

```diff
@@ -0,0 +1,147 @@
+        $this->assertSame($this->applicationType, $data['initiative']);
+        $this->assertSame($this->applicationUrl, $data['initiativeContext']);
+    }
+
+    /**
+     * @return void
+     */
+    public function testValidationUrl()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setValidationUrl($this->validationUrl));
+        $this->assertEquals($this->validationUrl, $request->getValidationUrl());
+    }
+
+    /**
+     * @expectedException        \Omnipay\Common\Exception\InvalidRequestException
+     * @expectedExceptionMessage The validationUrl parameter is required
+     * @return                   void
+     */
+    public function testValidationUrlRequired()
+    {
+        $this->request->setValidationUrl(null);
+        $this->request->getData();
+    }
+
+    /**
+     * @return void
+     */
+    public function testMerchantIdentifier()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setMerchantIdentifier($this->merchantIdentifier));
+        $this->assertSame($this->merchantIdentifier, $request->getMerchantIdentifier());
+    }
+
+    /**
+     * @return void
+     */
+    public function testDisplayName()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setDisplayName($this->displayName));
+        $this->assertSame($this->displayName, $request->getDisplayName());
+    }
+
+    /**
+     * @return void
+     */
+    public function testApplicationType()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setApplicationType($this->applicationType));
+        $this->assertSame($this->applicationType, $request->getApplicationType());
+    }
+
+    /**
+     * @return void
+     */
+    public function testApplicationUrl()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setApplicationUrl($this->applicationUrl));
+        $this->assertSame($this->applicationUrl, $request->getApplicationUrl());
+    }
+
+   /**
+	* @return void
+	*/
+    public function testSendSuccess()
+    {
+        $this->setMockHttpResponse('ApplePayAuthorizeRequestSuccess.txt');
+        $response = $this->request->send();
+
+        $this->assertTrue($response->isSuccessful());
+        $this->assertSame('OK', $response->getReason());
+        $this->assertSame(200, $response->getStatusCode());
+    }
```

Comment:
Adding this change now.

Reply:
doesn't look like this ever got added?

Reply:
oh i see you added this in ApplePayAuthorizeResponseTest.php. Omnipay typically just tests methods on the response in the RequestTests, because they have a lot of the same boilerplate and what's important is really that you can get the appropriate data after making the request. so i'd recommend just combining the two files.